### PR TITLE
Add Vanilla GAN Notebook tutorial

### PR DESCRIPTION
**Pull request by Arquestro**
## Description

Addition of tutorial for GANs in Notebook API.

## Related Issue

This is a clone of PR #653

## Type of Change

- [x] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have read I need to click 'Login as guest' to see Teamcity build logs
- [x] I have checked the code-style using `make check-codestyle`.
- [ ] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by Arquestro** [comment]
link - https://colab.research.google.com/github/catalyst-team/catalyst/blob/feature/gan-notebook-tutorial/examples/notebooks/vanilla-gan-tutorial.ipynb

File: examples/notebooks/vanilla-gan-tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,1000 @@
+{
```
Comment (reyzharkov) [inline examples/notebooks/vanilla-gan-tutorial.ipynb:1]: <p>1)you have already imported torch in the previous section</p><p>2)why should you import numpy by 2 different names?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-team/catalyst/pull/660/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (reyzharkov) [inline examples/notebooks/vanilla-gan-tutorial.ipynb:1]: <p>Can you explain what's happening here - what is this wrapping and why user needs it, and for which callbacks? Maybe explain the concept of phases</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-team/catalyst/pull/660/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (reyzharkov) [inline examples/notebooks/vanilla-gan-tutorial.ipynb:1]: <p>We are still waiting for <code>runner.infer</code>  to be implemented =)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-team/catalyst/pull/660/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (reyzharkov) [inline examples/notebooks/vanilla-gan-tutorial.ipynb:1]: <p>you can use <code>torchvision.utils.make_grid</code></p><p>(you can left everything as it is, it was just fyi)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-team/catalyst/pull/660/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

**Comment by reyzharkov** [comment]
It might be a good idea to pick a reviewer who had not yet an experience with catalyst gans instead of three of us. This may help to understand what is not really clear in this example from the first glance. @Scitator can you pick someone?

**Comment by Scitator** [comment]
@Arquestro do you have any plan for example update? we have a bunch of framework updates :)

**Comment by Scitator** [comment]
due to @asmekal proposal, this PR moving to [Catalyst.GAN](https://github.com/catalyst-team/gan)

**Pull request closed without merging**